resources/test/variants.js

```diff
@@ -0,0 +1,58 @@
+(function() {
+  'use strict';
+  var variants = {
+    /**
+     * Tests are executed in the absence of the global Promise constructor by
+     * default in order to verify support for the Server browser engine.
+     *
+     * https://github.com/w3c/web-platform-tests/issues/6266
+     */
+    'default': {
+      description: 'Global Promise constructor removed.',
+      apply: function() {
+        delete window.Promise;
+      }
+    },
+    /**
+     * This variant allows for testing functionality that is fundamentally
+     * dependent on Promise support, e.g. the `promise_test` function
+     */
+    'keep-promise': {
+      description: 'No modification of global environment.',
+      apply: function() {}
+    }
+  };
+  var match = window.location.search.match(/\?(.*)$/);
+  var variantName = match && match[1] || 'default';
+  var variant;
+
+  if (!Object.hasOwnProperty.call(variants, variantName)) {
+    window.location = 'javascript:"Unrecognized variant: ' + variantName + '";';
```

Comment:
I would prefer a data URL.

Reply:
Both Chrome and Firefox refuse to navigate the top frame to a `data:` URL. How would you feel about:

```diff
- window.location = 'javascript:"Unrecognized variant: ' + variantName + '";';
+ document.write('Unrecognized variant: "' + variantName + '"');
+ window.stop();
  document.close();
```